mailpoet/lib/Config/ServicesChecker.php

```diff
@@ -133,6 +133,14 @@ public function isPremiumKeyValid($displayErrorNotice = true) {
     return false;
   }
 
+  public function isBundledSubscription(): bool {
+    if ($this->isMailPoetAPIKeyValid(false, true) || $this->isPremiumKeyValid(false)) {
```

Comment:
If I understand correctly, in case we disable the key on the shop `isBundledSubscription` will start returning `false`. The shop may block the key due to overreaching limits or sending spam. With the current implementation on the Bridge, the `/me` and `/premium` endpoints will start returning 403 or another code and keys will not be marked as valid.

So for users, this could lead to suddenly seeing non-premium variants of the automation and Woo features.
Maybe we could drop the check here? That would, IMO mean asking for the last known state from when the key was valid.

Or another option could be using `$this->getValidAccountKey() !== null)`. The getValidAccountKey method checks if the key is valid or valid but underprivileged. Such a key doesn't grant access to MSS or Premium but still can be used e.g. for reporting stats, and Bridge knows it exists. So the `isBundledSubscription` would also work for blocked keys but not for spam issues because the shop currently blocks spammers on the account level and removes all keys from the bridge (this will hopefully change in the future). But still its slightly better than checking only valid states.